src/OnlineRetailPortal.Services/Translators/AddProductRequestTranslator.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using OnlineRetailPortal.Core;
+using System.Linq;
+
+namespace OnlineRetailPortal.Services
+{
+    public static class AddProductRequestTranslator
+    {
+        public static Product ToCore(this Contracts.AddProductRequest addProductRequest)
+        {
+            var product = new Product()
+            {
+                Id = addProductRequest.SellerId,
+                Name = addProductRequest.Name,
+                Description = addProductRequest.Description,
+                HeroImage = new Image { Url = addProductRequest.HeroImage.Url },
+                Price = new Price { Amount = addProductRequest.Price.Amount, IsNegotiable = addProductRequest.Price.IsNegotiable },
+                Category = (Category)addProductRequest.Category,
+                PurchasedDate = addProductRequest.PurchasedDate,
+                Images = addProductRequest.Images.Select(x => new Image
+                {
+                    Url = x.Url
+                }).ToList(),
+                PickupAddress = new Address
```

Comment:
please add null checks for optional fields

Reply:
done